Update buildx reference docs

thaJeztah opened the pull request:
relates to https://github.com/docker/buildx/pull/91 and https://github.com/docker/buildx/pull/506

This brings in more detailed command-line reference for the buildx subcommands; some enhancements were made to the template to add support for linking flag descriptions to a section on the page (or possibly an URL).

For example; the flag names in the "options" table on this page is now clickable; https://deploy-preview-12092--docsdocker.netlify.app/engine/reference/commandline/buildx_bake/#options

<img width="1007" alt="Screenshot 2021-03-26 at 16 58 37" src="https://user-images.githubusercontent.com/1804568/112659792-07410c80-8e55-11eb-8681-90b83885b35e.png">


Which allows users to click to scroll to the relevant section on the page, for example, the section describing the `--progress` flag; https://deploy-preview-12092--docsdocker.netlify.app/engine/reference/commandline/buildx_bake/#progress

thaJeztah commented:
@usha-mandya @tonistiigi @dieucao @chris-crone ptal

thaJeztah commented:
Pushed one more commit to update the "working with buildx" page

thaJeztah commented:
also added the missing pages for `docker buildx install` and `docker buildx uninstall` (these commands are hidden, but were referenced in the "work with buildx" introduction

usha-mandya commented on _data/buildx/docker_buildx_build.yaml:
```diff
@@ -1,7 +1,13 @@
+  The `buildx build` command starts a build using BuildKit. This command is similar
+  to the UI of `docker build` command and takes the same flags and arguments.
+
+  For documentation on most of these flags refer to the [`docker build`
```
```suggestion
  For documentation on most of these flags, refer to the [`docker build`
```

usha-mandya commented on _data/buildx/docker_buildx_build.yaml:
```diff
@@ -351,9 +369,228 @@ inherited_options:
+  built for all of the specified platforms and joined together into a single manifest
+  list.
+
+  If the`Dockerfile` needs to invoke the `RUN` command, the builder needs runtime
```
```suggestion
  If the `Dockerfile` needs to invoke the `RUN` command, the builder needs runtime
```

usha-mandya commented on _data/buildx/docker_buildx_build.yaml:
```diff
@@ -351,9 +369,228 @@ inherited_options:
+  support for the specified platform. In a clean setup, you can only execute `RUN`
+  commands for your system architecture.
+  If your kernel supports [`binfmt_misc`](https://en.wikipedia.org/wiki/Binfmt_misc)
+  launchers for secondary architectures buildx will pick them up automatically.
```
```suggestion
  launchers for secondary architectures, buildx will pick them up automatically.
```

usha-mandya commented on _data/buildx/docker_buildx_build.yaml:
```diff
@@ -351,9 +369,228 @@ inherited_options:
+  commands for your system architecture.
+  If your kernel supports [`binfmt_misc`](https://en.wikipedia.org/wiki/Binfmt_misc)
+  launchers for secondary architectures buildx will pick them up automatically.
+  Docker desktop releases come with binfmt_misc automatically configured for `arm64`
```
```suggestion
  Docker desktop releases come with `binfmt_misc` automatically configured for `arm64`
```

usha-mandya commented on _data/buildx/docker_buildx_create.yaml:
```diff
@@ -1,6 +1,15 @@
+  where context is the name of a context from `docker context ls` and endpoint is
+  the address for docker socket (eg. `DOCKER_HOST` value).
+
+  By default, the current docker configuration is used for determining the
```
```suggestion
  By default, the current Docker configuration is used for determining the
```

usha-mandya commented on _data/buildx/docker_buildx_create.yaml:
```diff
@@ -1,6 +1,15 @@
+  context/endpoint value.
+
+  Builder instances are isolated environments where builds can be invoked. All
+  docker contexts also get the default builder instance.
```
```suggestion
  Docker contexts also get the default builder instance.
```

usha-mandya commented on _data/buildx/docker_buildx_imagetools_create.yaml:
```diff
@@ -1,6 +1,13 @@
+  Imagetools contains commands for working with manifest lists in the registry.
+  These commands are useful for inspecting multi-platform build results.
+
+  Create creates a new manifest list based on source manifests. The source
```
```suggestion
  Create a new manifest list based on source manifests. The source
```

usha-mandya commented on _data/buildx/docker_buildx_imagetools_create.yaml:
```diff
@@ -1,6 +1,13 @@
+  Create creates a new manifest list based on source manifests. The source
+  manifests can be manifest lists or single platform distribution manifests and
+  must already exist in the registry where the new manifest is created. If only
+  one source is specified create performs a carbon copy.
```
```suggestion
  one source is specified, create performs a carbon copy.
```

usha-mandya commented on _data/buildx/docker_buildx_inspect.yaml:
```diff
@@ -23,9 +24,43 @@ inherited_options:
+
+  ### Ensure that the builder is running before inspecting (--bootstrap) {#bootstrap}
+
+  Use the `--bootstrap` option to ensures that the builder is running before
```
```suggestion
  Use the `--bootstrap` option to ensure that the builder is running before
```

usha-mandya commented on _data/buildx/docker_buildx_inspect.yaml:
```diff
@@ -23,9 +24,43 @@ inherited_options:
+  Use the `--bootstrap` option to ensures that the builder is running before
+  inspecting it. If the driver is `docker-container`, then `--bootstrap` starts
+  the buildkit container and waits until it is operational. Bootstrapping is
+  automatically done during build, it is thus not necessary. The same BuildKit
```
```suggestion
  automatically done during build, and therefore not necessary. The same BuildKit
```

usha-mandya commented on buildx/working-with-buildx.md:
```diff
@@ -4,84 +4,150 @@ description: Working with Docker Buildx
-You can run Buildx in different configurations that are exposed through a driver concept. Currently, Docker supports a "docker" driver that uses the BuildKit library bundled into the docker daemon binary, and a "docker-container" driver that automatically launches BuildKit inside a Docker container.
+You can run Buildx in different configurations that are exposed through a driver
+concept. Currently, Docker supports a "docker" driver that uses the BuildKit
+library bundled into the docker daemon binary, and a "docker-container" driver
```
```suggestion
library bundled into the Docker daemon binary, and a "docker-container" driver
```

usha-mandya commented on buildx/working-with-buildx.md:
```diff
@@ -4,84 +4,150 @@ description: Working with Docker Buildx
+To list all available builders, use [`docker buildx ls`](../engine/reference/commandline/buildx_ls.md].
+After creating a new builder you can also append new nodes to it.
+
+To switch between different builders use [`docker buildx use <name>`](../engine/reference/commandline/buildx_use.md).
```
```suggestion
To switch between different builders, use [`docker buildx use <name>`](../engine/reference/commandline/buildx_use.md).
```

usha-mandya commented on buildx/working-with-buildx.md:
```diff
@@ -4,84 +4,150 @@ description: Working with Docker Buildx
 
-Finally, depending on your project, the language that you use may have good support for cross-compilation. In that case, multi-stage builds in Dockerfiles can be effectively used to build binaries for the platform specified with `--platform` using the native architecture of the build node. A list of build arguments like `BUILDPLATFORM` and `TARGETPLATFORM` is available automatically inside your D ...
+Finally, depending on your project, the language that you use may have good 
+upport for cross-compilation. In that case, multi-stage builds in Dockerfiles
```
```suggestion
support for cross-compilation. In that case, multi-stage builds in Dockerfiles
```

usha-mandya commented on _data/buildx/docker_buildx_build.yaml:
```diff
@@ -351,9 +369,228 @@ inherited_options:
+  #### `image`
+
+  The `image` exporter writes the build result as an image or a manifest list. When
+  using `docker` driver the image will appear in `docker images`. Optionally image
```
```suggestion
  using `docker` driver, the image will appear in `docker images`. Optionally, image
```

usha-mandya commented on _data/buildx/docker_buildx_create.yaml:
```diff
@@ -33,6 +45,7 @@ options:
 - option: driver
   value_type: string
   description: 'Driver to use (available: [])'
```
Should we use a backtick here 

![image](https://user-images.githubusercontent.com/47779042/112885904-2e077900-90c9-11eb-942c-b485b256e2ff.png)

thaJeztah replied:
This description is coming from the generated code, and on the command-line we don't print backtics, so I'll need to have a look at this separately (we have the same issue on some of our other reference docs pages)

usha-mandya commented:
@thaJeztah Thanks for the PR. Looks good overall. Left some minor suggestions.

thaJeztah commented:
@usha-mandya I addressed your comments; I have a PR pending upstream with those changes (https://github.com/docker/buildx/pull/583) but already regenerated the yaml files with those applied and updated this PR (so should be ready for review/merge)

usha-mandya reviewed (approved):
LGTM. Thank you @thaJeztah

thaJeztah commented:
Thanks!
